splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceWatcher.java

```diff
@@ -160,21 +173,61 @@ public PreparedStatement prepareStatement(String sql) throws SQLException {
         return ps;
     }
 
+    /**
+     * Try closing connections gracefully from different threads, if that takes long it could be some queries are stuck,
+     * so abort the connections forcefully instead
+     */
     private void closeConnections() {
+        int numThreads = connections.size() + 2; // two extra for the .abort() call
+        ExecutorService executor = Executors.newFixedThreadPool(numThreads,
```

Comment:
The thread pool is shared right? according to the doc:
> public static ExecutorService newFixedThreadPool(int nThreads)
> ... At any point, at most nThreads threads will be active processing tasks. If additional tasks are submitted when all threads are active, they will wait in the queue until a thread is available.

If we run Test suites in parallel which use `SpliceWatcher`, could it lead to competition between different suites calling `closeConnection` on parallel and causing some of them to time out (because of line 198)?

Reply:
It wasn't shared, but I think it's better if it is. I made the Executor a cachedThreadPool without a thread limit, so resources should be enough for concurrent ITs.